examples/log-mpi.py

```diff
@@ -45,8 +45,10 @@ def main():
     logmgr.add_quantity(Fifteen("fifteen"))
 
     # Watches are printed periodically during execution
-    logmgr.add_watches(
-        ["step.max", "t_sim.max", "t_step.max", "fifteen", "t_vis.max"])
+    logmgr.add_watches(["step.max", "t_step.min", "t_step.max", "t_sim.max",
+                        "fifteen", "t_vis.max"])
+
+    logmgr.set_watch_format(["step={value} ", "\nt_step({value:g},", "{value:g})\n"])
```

Comment:
I find this hard to use. Why not set the watch format at the time when watches are added? Like so maybe:
```
[("step.max", "step={value}"), ...]
```

Reply:
The (slight) issue here is that `add_watches` already accepts a tuple as each list element, with an unspecified `expr` as the second tuple item:

https://github.com/illinois-ceesd/logpyle/blob/84fba7674f0ed72e018016fe15eb50fbad45507f/logpyle/__init__.py#L612-L620

Not sure if it's worth complicating this interface further.

Reply:
We could introduce a dataclass to hold all those bits of information. Also, it looks like this formatting effectively subsumes the previous "label" mechanism. We shouldn't have two bits of functionality to do the same thing.

Reply:
> We could introduce a dataclass to hold all those bits of information. 

I'm not sure I understand. Which bits are you referring to? Information about watches are already stored in a `pytools.Record`. I think we can't rely on dataclasses, since they are a 3.7+ feature and we support 3.6.

> Also, it looks like this formatting effectively subsumes the previous "label" mechanism. We shouldn't have two bits of functionality to do the same thing.

What is the label mechanism you are referring to?

Reply:
> I think we can't rely on dataclasses, since they are a 3.7+ feature and we support 3.6.

There's a backport. See its use in mirgecom e.g.

> Which bits are you referring to?

- Labeling
- Expression
- Formatting

What is the label mechanism you are referring to?

The `display` part above serves as a label AFAIK.

Reply:
> > I think we can't rely on dataclasses, since they are a 3.7+ feature and we support 3.6.
> 
> There's a backport. See its use in mirgecom e.g.

So, you suggest to modify `add_watches` so that it receives a list of dataclasses from the user? With the following elements:

- expr (required)
- unit (optional)
- format (optional)
?

Reply:
> > > I think we can't rely on dataclasses, since they are a 3.7+ feature and we support 3.6.
> > 
> > 
> > There's a backport. See its use in mirgecom e.g.
> 
> So, you suggest to modify `add_watches` so that it receives a list of dataclasses from the user? With the following elements:
> 
> * `expr` (required)
> * ~`unit` (optional)~ Edit: maybe not required, could be part of format
> * `format` (optional)
> 
> ?
> 
> Is the explicit `expr` in use anywhere? If not, we could just reuse the exisiting API of `add_watches` to receive either an `expr` scalar or `(expr, format)` tuple (without needing a dataclass).

I implemented this.